flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/TaskIOMetricGroup.java

```diff
@@ -196,6 +204,23 @@ public void markTaskStart() {
         this.taskStartTime = System.currentTimeMillis();
     }
 
+    public void markTaskInitialize() {
+        this.taskInitializeTime = System.currentTimeMillis();
```

Comment:
It's not trivial to follow all the code paths inside `StreamTask` that could lead to this call (`restoreInternal` can be called from `invoke`).
Should we make this update conditional - i.e. only update the field if it wasn't already updated?

Reply:
`isRunning` can help us [check](https://github.com/apache/flink/blob/master/flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskTest.java#L1423) whether task has been restored, should we check this conditional again?